src/utils/replace.rs

```diff
@@ -0,0 +1,421 @@
+// Copyright 2020 Christopher Sugai.
+// Licensed under the MIT license (http://opensource.org/licenses/MIT)
+// This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! 
+//! Traits to replace non-ACTG nucleotide characters with pseudorandom bases. Additional functionality to create strings of pseudorandom nucleotides and convert to uppercase.
+//! # Example
+//! ```
+//! 
+//! extern crate bio;
+//! extern crate rand;
+//! 
+//! use rand::rngs::ThreadRng;
+//! use std::string::String;
+//! use rand::seq::SliceRandom;
+//! use bio::utils::TextSlice;
+//! use bio::utils::Text;
+//! 
+//! let mut rng = rand::thread_rng(); //create a random number generator
+//! let x1 = "ACTGuyUNn.-@<^>".to_string(); //string
+//! let x2 = "ACTGuyUNn.-@<^>"; //stringslice
+//! let x3: TextSlice = "ACTGuyUNn.-@<^>".as_bytes(); //rust bio textslice
+//! let x4: Text = "ACTGuyUNn.-@<^>".as_bytes().to_vec(); //rust bio text
+//!
+//!
+//! let y1 = x1.replace_gap(rng); //replace gap {.-} using random number generator
+//!
+//! let y2 = x2.to_string(); //convert stringslice to either Text or String but must be owned
+//! let y2 = y2.replace_n(rng); //replace ambiguous/uncalled base {nN}
+//!
+//! let y3 = std::str::from_utf8(x3).expect("Not valid UTF8").to_string(); //convert textslice to either Text or String but must be owned
+//! let y3 = y3.replace_u_with_t(); //replace all other text other than ACTGUactgu, does not take rng
+//! 
+//! let y4 = x4.replace_all(rng); //replace u with t
+//! let y4 = std::string::String::from_utf8(y4).expect("Not valid UTF8");
+//!
+//! assert_eq!(y4.len(), "ACTGuyUNn.-@<^>".to_string().len());
+//!
+//! ```
+//! 
+//! 
+//! 
+//! 
+
+extern crate bio;
+use rand::rngs::ThreadRng;
+use std::string::String;
+use rand::seq::SliceRandom;
+
+use bio::utils::TextSlice;
+use bio::utils::Text;
+
+trait ReplaceNucleotide <T: ToOwned> {
+    fn replace_u_with_t(&self) -> Self;
+    fn replace_gap(&self, rng: ThreadRng) -> Self;
+    fn replace_gap_lowercase(&self, rng: ThreadRng) -> Self;
+    fn replace_n(&self, rng: ThreadRng) -> Self;
+    fn replace_iupac(&self, rng: ThreadRng) -> Self;
+    fn replace_lowercase_with_uppercase(&self, rng: ThreadRng) -> Self;
```

Comment:
Should be better named `to_uppercase`, since `std::str` has this method. It is idiomatic in Rust to reuse naming conventions in std